service/mattermost/mattermost_test.go

```diff
@@ -97,3 +99,27 @@ func TestService_Send(t *testing.T) {
 	assert.NotNil(err)
 	mockClient.AssertExpectations(t)
 }
+
```

Comment:
Even if these methods do not return values, it's still worth while asserting that the hooks work. To me, it's important to have at least one assertion per test. Here are some ideas you can take or leave when testing the pre and post hook:
1. assert that a pre-send function fires before a message is sent.
2. if two pre-hooks are registered and the first hook errors, the second hook is not called.
3. assert that the post send function after the a message is sent.

Just like your other tests, using a mockHttpClient should prove useful for these tests. Feel free to reach out for help if you're stuck.

Reply:
I have added basic assertions using mockHttpClient as suggested, please have a look.

Reply:
This is a good start. I see where this is going but we want to make an assertion on the behavior of the hook. At minimum these test should fail if the hook isn't called when a message is sent and should be calling the send method.

Reply:
I have added `mockClient.AssertExpectations(t)` along with calling `service.Send()` method.

I'm assuming that you are suggesting checking if the httpClient (i.e., `mockClient.` not `service.`) pre-send and post-send hooks are called before and after send method call respectively. Not sure if we can do this since generated mockClient doesn't call (is not aware) these hooks in the generated mockClient.Send().

please suggest how this could be achieved if possible, that would be great for testing complete capabilities.